ingest/src/main/java/com/microsoft/azure/kusto/ingest/ManagedStreamingIngestClient.java

```diff
@@ -0,0 +1,132 @@
+package com.microsoft.azure.kusto.ingest;
+
+import com.microsoft.azure.kusto.data.ConnectionStringBuilder;
+import com.microsoft.azure.kusto.data.StreamingClient;
+import com.microsoft.azure.kusto.ingest.exceptions.IngestionClientException;
+import com.microsoft.azure.kusto.ingest.exceptions.IngestionServiceException;
+import com.microsoft.azure.kusto.ingest.result.IngestionResult;
+import com.microsoft.azure.kusto.ingest.source.BlobSourceInfo;
+import com.microsoft.azure.kusto.ingest.source.FileSourceInfo;
+import com.microsoft.azure.kusto.ingest.source.ResultSetSourceInfo;
+import com.microsoft.azure.kusto.ingest.source.StreamSourceInfo;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.net.URISyntaxException;
+
+/**
+ * <p>ManagedStreamingIngestClient</p>
+ * <p>
+ * This class combines a managed streaming client with a queued streaming client, to create an optimized experience.
+ * Since the streaming client is communicating directly with the engine, it's more prone to failure, so this class
+ * holds both a streaming client and a queued client.
+ * At first it tries using the streaming client, and if it fails retries it {@value MAX_RETRY_CALLS} times, and if that
+ * still fails it then uses the queued streaming client.
```

Comment:
```suggestion
 * It tries {@value MAX_RETRY_CALLS} times using the streaming client, after which it falls back to the queued streaming client in case of failure.
```